packages/loader/container-loader/src/retriableDocumentStorageService.ts

```diff
@@ -89,17 +109,40 @@ export class RetriableDocumentStorageService implements IDocumentStorageService
                 if (!canRetryOnError(err)) {
                     throw err;
                 }
+                numRetries += 1;
+                lastError = err;
                 // If the error is throttling error, then wait for the specified time before retrying.
                 // If the waitTime is not specified, then we start with retrying immediately to max of 8s.
                 retryAfter = getRetryDelayFromError(err) ?? Math.min(retryAfter * 2, 8000);
+                totalWaitTime += retryAfter;
```

Comment:
I'd not sum up times, as this excludes time waiting for fetch() itself.
I'd just capture current time at the start of function, and do diff with current time at end of function, and use it to report total time to execute a call

Reply:
done